messaginginapp/src/main/java/io/customer/messaginginapp/ui/controller/InAppMessageViewController.kt

```diff
@@ -1,89 +1,86 @@
-package io.customer.messaginginapp.ui
+package io.customer.messaginginapp.ui.controller
 
 import android.content.ActivityNotFoundException
-import android.content.Context
-import android.util.AttributeSet
-import android.widget.FrameLayout
-import androidx.annotation.AttrRes
-import androidx.annotation.StyleRes
+import android.view.ViewGroup
 import androidx.annotation.UiThread
 import io.customer.messaginginapp.di.inAppMessagingManager
 import io.customer.messaginginapp.gist.data.model.Message
 import io.customer.messaginginapp.gist.data.model.engine.EngineWebConfiguration
 import io.customer.messaginginapp.gist.presentation.engine.EngineWebView
 import io.customer.messaginginapp.gist.presentation.engine.EngineWebViewListener
 import io.customer.messaginginapp.state.InAppMessagingAction
-import io.customer.messaginginapp.state.InAppMessagingState
 import io.customer.messaginginapp.type.InAppMessage
 import io.customer.messaginginapp.type.InlineMessageActionListener
+import io.customer.messaginginapp.ui.bridge.InAppMessageViewListener
 import io.customer.messaginginapp.ui.bridge.InAppPlatformDelegate
 import io.customer.sdk.core.di.SDKComponent
 
 /**
- * Base view class for displaying in-app messages. This class is responsible for managing common
- * in-app message functionality, such as setting up the EngineWebView, handling tap events, etc.
- * This class will be extended by specific in-app message views (e.g., ModalInAppMessageView)
- * to handle specific message types.
+ * Base controller class for managing in-app message view behavior.
+ * Encapsulates logic for view state transitions, sizing, lifecycle, and WebView interaction.
+ * Designed to decouple business logic from Android view classes for better testability and reuse.
  */
-abstract class InAppMessageBaseView @JvmOverloads constructor(
-    context: Context,
-    attrs: AttributeSet? = null,
-    @AttrRes defStyleAttr: Int = 0,
-    @StyleRes defStyleRes: Int = 0
-) : FrameLayout(context, attrs, defStyleAttr, defStyleRes), EngineWebViewListener {
-    internal abstract val platformDelegate: InAppPlatformDelegate
-    internal var engineWebView: EngineWebView? = null
-
-    protected var currentMessage: Message? = null
-    protected var currentRoute: String? = null
+internal abstract class InAppMessageViewController<ViewCallback : InAppMessageViewListener>(
+    protected val type: String,
+    protected val platformDelegate: InAppPlatformDelegate,
+    protected val viewDelegate: ViewGroup
+) : EngineWebViewListener {
+    internal val logger = SDKComponent.logger
+    internal val inAppMessagingManager = SDKComponent.inAppMessagingManager
+
+    internal var engineWebViewDelegate: EngineWebView? = null
```

Comment:
`EngineWebView` is also replaced with delegate in next PR in stack to decouple controller.